mlflow/tracking/client.py

```diff
@@ -1075,6 +1075,128 @@ def _is_plotly_figure(fig):
             else:
                 raise TypeError("Unsupported figure object type: '{}'".format(type(figure)))
 
+    @experimental
+    def log_image(self, run_id, image, artifact_file):
+        """
+        Log an image as an artifact. The following image objects are supported:
+
+        - `numpy.ndarray`_
+        - `PIL.Image.Image`_
+
+        .. _numpy.ndarray:
+            https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html
+
+        .. _PIL.Image.Image:
+            https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image
+
+        :param run_id: String ID of the run.
+        :param image: Image to log.
+        :param artifact_file: The run-relative artifact file path in posixpath format to which
+                              the figure is saved (e.g. "dir/image.png").
+
+        .. code-block:: python
+            :caption: Numpy Example
+
+            import mlflow
+            import numpy as np
+
+            image = np.random.randint(0, 256, size=(100, 100, 3), dtype=np.uint8)
+
+            run = client.create_run(experiment_id="0")
+            client.log_image(run.info.run_id, image, "image.png")
+
+        .. code-block:: python
+            :caption: Pillow Example
+
+            import mlflow
+            from PIL import Image
+
+            image = Image.new("RGB", (100, 100))
+
+            run = client.create_run(experiment_id="0")
+            client.log_image(run.info.run_id, image, "image.png")
+        """
+
+        def _is_pillow_image(image):
+            from PIL.Image import Image
+
+            return isinstance(image, Image)
+
+        def _is_numpy_array(image):
+            import numpy as np
+
+            return isinstance(image, np.ndarray)
+
+        def _normalize_to_uint8(x):
+            # Based on: https://github.com/matplotlib/matplotlib/blob/06567e021f21be046b6d6dcf00380c1cb9adaf3c/lib/matplotlib/image.py#L684
+
+            is_int = np.issubdtype(x.dtype, np.integer)
+            low = 0
+            high = 255 if is_int else 1
+            if x.min() < low or x.max() > high:
+                _logger.warning(
+                    "Clipping array (dtype: '{}') to [{}..{}]".format(x.dtype, low, high)
+                )
```

Comment:
I started thinking we should raise an exception instead of a warning because if a user turned off warnings (with `warnings.filterwarnings("ignore")`), he wouldn't notice that the array was clipped.

Reply:
I often see people turn off warnings **on a notebook** because they are distracting.